integration-tests/desugar/type-annotation-check-varname-err-not-same/test.test.js

```diff
@@ -0,0 +1,10 @@
+const test = require('ava');
+const {exec} = require('../../setup');
+
+test.failing(
+    "desugar: type annotation var name doesn't agree with the declaration var name",
+    exec,
+    __dirname,
+    ["-m", "src/Main.elm"],
+    async program => ({snapshot: await program})
```

Comment:
On the back of #105 I think something more like the test in https://github.com/elm-in-elm/compiler/blob/1c8669fb0e7907ed00cd2f6d854803f2f67825a9/integration-tests/errors/missing-equals/test.test.js would avoid the need for `.failing` here.

```suggestion
    async (program, t) => {
        const snapshot = await t.throwsAsync(program);
        t.is(snapshot.code, 1);
        return snapshot;
    }
```

Reply:
Thanks Harry! I just wanted to ping you here :)

I'll try to incorporate that suggestion to all the  *-err-* tests here when I get to my PC.

Reply:
I think with the magic I have put in the [exec macro](https://github.com/elm-in-elm/compiler/blob/1c8669fb0e7907ed00cd2f6d854803f2f67825a9/integration-tests/setup.js) I have made the mistake of reducing boiler plate code at the cost of code readability.

Maybe I should look at making this macro a bit less magic. (For reference test macros are described [here](https://github.com/avajs/ava/blob/19c4f35fbae740268dedeeb4d450f813f2d5e85e/docs/01-writing-tests.md#reusing-test-logic-through-macros)).